src/usePagenation/createPagenation.ts

```diff
@@ -0,0 +1,87 @@
+
+
+interface Pagenation {
+    current: number;
+    pageSize: number;
+    total: number;
+    data: any[] | null;
+    onChange: any;
+    run: any;
+}
+
+export default class CreatePagenation {
+    current: number;
+    pageSize: number;
+    total: number;
+    request: (params: any) => Promise<any>;
+    options: any;
+    updateFunc: any;
+    data: any[] | null;
+    loading: boolean;
+    constructor({ request, options = {}, updateFunc }: any) {
+        this.current = options.defaultcurrent || 1;
+        this.pageSize = options.defaultPageSize || 20;
+        this.total = 0;
+        this.request = request;
+        this.options = options;
+        this.updateFunc = updateFunc;
+        this.data = [];
+        this.loading = false;
+    }
+
+    getData = (params: any = {}) => {
+        const { dataKey = 'data', currentKey = 'size', sizeKey = 'page', totalKey = 'total' } = this.options;
+        const newParams: any = Object.assign(params, {
+            [currentKey]: this.current,
+            [sizeKey]: this.pageSize,
+        });
+        const updateFunc = this.updateFunc;
+        // const that = this;
```

Comment:
Probably going with hooks/functions instead of a class would make things easier. I can't suggest a proper implementation but it looks like you're implementing Java like OOP in TS.

Reply:
look look this, it's a lot easier.